indykite/resource_authorization_policy.go

```diff
@@ -74,60 +86,43 @@ func resourceAuthorizationPolicyFlatten(
 	resp *configpb.ReadConfigNodeResponse,
 ) (d diag.Diagnostics) {
 	policy := resp.GetConfigNode().GetAuthorizationPolicyConfig().GetPolicy()
-	if policy == nil {
+	if policy == "" {
 		return append(d, buildPluginError("config in the response is not valid AuthorizationPolicyConfig"))
 	}
+	setData(&d, data, authorizationPolicyJSONConfigKey, policy)
 
-	jsonVal, err := protojson.MarshalOptions{EmitUnpopulated: true, UseEnumNumbers: true}.Marshal(policy)
-	if err != nil {
-		return append(d, buildPluginError("failed to marshall message into JSON: "+err.Error()))
+	status := resp.GetConfigNode().GetAuthorizationPolicyConfig().GetStatus()
+	if status == configpb.AuthorizationPolicyConfig_STATUS_INVALID {
+		return append(d, buildPluginError("status in the response is not valid"))
 	}
+	setData(&d, data, statusKey, status.String())
```

Comment:
If Proto is changed, but we don't change the enum, it will cause troubles. It will set invalid value into status. 

Handle it with existence checks, like this:
https://github.com/indykite/terraform-provider-indykite/blob/master/indykite/resource_oauth2_client.go#L159-L163